File: Java/programas/Anios, Meses, Semanas, Dias/src/DiasYMeses.java
import java.io.IOException; //Excepción si hay error durante entrada y salida de datos.
import java.util.InputMismatchException; //Excepción si hay un error de formato en tipos de datos.
import java.util.Scanner; //Librería para declara objeto Scanner de entrada y salida

public class DiasYMeses {
    private int Anios, Dias, Semanas, Meses; //Atributos privados.
    Scanner lectura = new Scanner(System.in); //Declara objeto Scanner.

    //Procedimiento que da comienzo al programa.
    public void Iniciar() throws InterruptedException, IOException{
        new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor(); //Limpia consola/terminal/pantalla.
        
        //Selectiva try-catch por si se ingresa un tipo de dato incorrecto en la variable
        try{
            System.out.printf("Ingresa un número entero para trabajarlo como días: "); Dias = lectura.nextInt();
    
            Calculo(); //Llamada al método que asigna valores a los atributos privados.
        } catch (InputMismatchException e){
            System.out.println("Error de formato.");
            lectura.nextLine(); //Limpia buffer de entrada de datos.
            Iniciar(); //Llamada recursiva a Iniciar.
        }
    }

    //Función que calcula valores para los atributos de la clase, relacionados con los años, meses y días transcurridos
    //en n días.
    private void Calculo(){
        Anios = Dias / 365;
        Meses = Dias / 30;
        Semanas = (Dias % 30) / 7;
    }

    //Procedimiento que despliega los atributos de la clase.
    public void Despliegue(){
        System.out.printf("%s días son: %s años, %s meses, %s semanas y %s días.", Dias, Anios, Meses, Semanas, (Dias % 30) % 7);
    }
}

File: C++/programas/P-Anios, Meses, Semanas, Dias/P-Anios, Meses, Semanas, Dias/DiasYMeses.cpp
#include "DiasYMeses.h" //Cabecera de clase DiasYMeses.
#include <iostream> //Cabecera para entrada y salida de datos.

//Procedimiento que calcula las semanas, meses y a?os de un n?mero que representa d?as.
void DiasYMeses::Calculo()
{
	Anios = Dias / 365;
	Meses = Dias / 30;
	Semanas = (Dias % 30) / 7;
}

//Procedimiento que recibe un n?mero entero que representa d?as.
void DiasYMeses::Iniciar()
{
	system("cls");

	std::cout << "Ingresa un n?mero entero para trabajarlo como d?as: "; std::cin >> Dias;

	if (std::cin.fail()) //Si se ingres? un tipo de dato incorrecto en la variable.
	{
		std::cout << "Error de formato.\n";
		system("pause");
		std::cin.clear(); //Limpia el buffer de cin.
		//Ignora el resto de caracteres ingresados en cin hasta el pr?ximo salto de l?nea.
		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
		Iniciar(); //Llamada recursiva a Iniciar.
	}
	else
		Calculo(); //Llamada a procedimiento que calcula componentes de un n?mero que representa d?as/
}

//Procedimiento que despliega los a?os, meses y semanas de una cantidad de d?as transcurridos.
void DiasYMeses::Despliegue()
{
	std::cout << Dias << " d?as son: " << Anios << " a?os, " << Meses << " meses, " << Semanas << " semanas y " << (Dias % 30 / 7) << " d?as.";
}